src/block_cache/block_mgr.c

```diff
@@ -18,16 +18,27 @@ int
 __wt_bm_close_block(WT_SESSION_IMPL *session, WT_BLOCK *block)
 {
     WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
 
     conn = S2C(session);
 
     __wt_verbose(session, WT_VERB_BLKCACHE, "close: %s", block->name);
 
+    /* We shouldn't have any read requests in progress. */
+    WT_ASSERT(session, block->read_count == 0);
+
     __wt_spin_lock(session, &conn->block_lock);
     if (block->ref > 0 && --block->ref > 0) {
```

Comment:
I know it's not in the changed code, but this feels odd to me.  Should we assert that `block->ref > 0` and skip the first part of the test in the `if`?  Question more for @keitharnoldsmith.

Reply:
I was trying to reference line 31, but it didn't highlight it:
```
if (block->ref > 0 && --block->ref > 0) {
```
That's the line that felt odd.

Reply:
@ddanderson I believe it is wrap-around protection on the unsigned `ref`.

Reply:
@marcbutler, okay.  But wouldn't `block->ref` being 0 at this point be totally unexpected?  I think we have to had gotten a reference to the block if we're going to call this function.

Reply:
@ddanderson I agree it is unexpected, and as such it feels more like an assertion. Why the excess of caution in the conditional expression, perhaps @keitharnoldsmith has some insight.

Reply:
I agree that it causes a bit of confusion because the names are so similar. But `block->read_count` is tracking something different -- the number of read requests to the `WT_BLOCK` that are currently active. So that assert is saying that we shouldn't be closing the block if there are active requests. See WT-10838.

Reply:
Regarding the odd conditional with `block->ref` I don't have any additional insight, other than the fact that it's been that way for quite a while. Earlier versions of the code say `if (block->ref == 0 || --block->ref == 0) ... close block`. So clearly the condition has been inverted since we now exit on the condition and close the block on the fall through. 

I agree with @marcbutler that it's probably protection against underflow. I would guess it's there so that if we hit this case (i.e., `block->ref == 0`) in a deployed system we do something reasonable. 

I agree with the logic that if we're closing the handle here we *should* have a positive reference count. So I don't see a harm in adding an assert to that effect. If something immediately breaks it would tell us something.

Reply:
To be clear, I'm okay with leaving this existing code for now - I'm not advocating making the change in this PR.  It just seemed like a wart that we should fix at some point.